Sources/Rego/IREvaluator.swift

```diff
@@ -2,6 +2,119 @@ import AST
 import Foundation
 import IR
 
+// Custom array type that supports IR.Local indexing
+internal struct Locals: Equatable, Sendable, Encodable {
+    private var storage: Array<AST.RegoValue?>
+
+    init() {
+        self.storage = []
+    }
+
+    init(_ elements: [AST.RegoValue?]) {
+        self.storage = elements
+    }
+
+    init(repeating value: AST.RegoValue?, count: Int) {
+        self.storage = Array(repeating: value, count: count)
+    }
+
+    init(accommodating locals: [IR.Local], minimumSize: Int = 0) {
+        let maxLocal = locals.max() ?? IR.Local(0)
```

Comment:
❓`Array.max()` is O(N). Will that be appropriate for the main use cases of this constructor?